.ci/install-guest-component-dependencies.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+#
+# Copyright Confidential Containers Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Detect if the OS is Ubuntu or fedora
+source "/etc/os-release" || source "/usr/lib/os-release"
+
+# On Ubuntu or debian, install the dependencies
+if [ "$ID" == "ubuntu" ] || [ "$ID" == "debian" ]; then
+    export DEBIAN_FRONTEND=noninteractive
+    sudo apt update
+    sudo apt install -y \
+        libdevmapper-dev \
+        clang
+
+    exit 0
+fi
+
+# On Fedora, install the dependencies
+if [ "$ID" == "fedora" ]; then
+    sudo dnf install -y \
+        device-mapper-devel \
+        clang
+
+    exit 0
+fi
+
+# On CentOS 8 Stream
+if [ "$ID" == "centos" ] && [ "$VERSION_ID" == "8" ]; then
+    # Enable powertools repo to be able to install the developer library for device-mapper
+    yum install -y \
+        dnf-plugins-core
+    yum config-manager --set-enabled powertools
+
+    yum install -y \
+        device-mapper-devel \
+        clang
```

Comment:
@ChengyuZhu6 done. Note that there is no sudo because when I was trying out in a container image of centos stream `quay.io/centos/centos:stream8` there was no sudo in there.

Reply:
Thanks for your messages. Let me take a look.